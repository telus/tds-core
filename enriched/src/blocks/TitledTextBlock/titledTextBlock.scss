
.titled-text-block {
  margin: 60px 0;

  h1 {
    margin-bottom: 46px;
  }
}

.title-text-block__content {
 margin: 32px 0;
}