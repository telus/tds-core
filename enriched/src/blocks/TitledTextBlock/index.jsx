import React, { Component } from 'react';

if ( process.env.BROWSER ) {
  require('./titledTextBlock.scss');
}

class TitledTextBlock extends Component {

  render() {
    const { title, content, titleTagType } = this.props;

    const contentHtml = content.map((c, index) => (
      <div key={index} className="titled-text-component">
        <h4>{c.title}</h4>
        <p>{c.content}</p>
      </div>
    ));

    return (
      <div className="titled-text-block">
        <div className="container container--limited-width">
          <div className="grid-row">
            <div className="small-12 medium-12 large-12">
              {React.createElement(titleTagType, null, title)}
              {contentHtml}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

TitledTextBlock.propTypes = {
  title: React.PropTypes.string.isRequired,
  content: React.PropTypes.arrayOf(React.PropTypes.shape({
    title: React.PropTypes.string.isRequired,
    content: React.PropTypes.string.isRequired
  })).isRequired,
  titleTagType: React.PropTypes.string
};

TitledTextBlock.defaultProps = {
  titleTagType: 'h1'
};

export default TitledTextBlock;
