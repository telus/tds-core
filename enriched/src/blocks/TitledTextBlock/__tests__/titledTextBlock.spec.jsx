import React from 'react';
import {mount} from 'enzyme';
import {mountToJson} from 'enzyme-to-json';
import TitledTextBlock from '../';

describe('TextBlock Component', () => {
  const data = {
    title: 'Title1',
    content: [
      {title: 'Small Title 1', content: 'Small Text1'}
    ]
  };

  it('matches snapshot', () => {
    const wrapper = mount(<TitledTextBlock {...data} />);
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should render title as default tag type', () => {
    const wrapper = mount(<TitledTextBlock {...data}/>);
    expect(wrapper.find('h1').first().render().text()).toEqual('Title1');
  });

  it('should render title as specified tag type', () => {
    const wrapper = mount(<TitledTextBlock {...data} titleTagType="h2" />);
    expect(wrapper.find('h2').first().render().text()).toEqual('Title1');
  });

  it('should render content', () => {
    const wrapper = mount(<TitledTextBlock {...data}/>);
    expect(wrapper.find('h4').first().render().text()).toEqual('Small Title 1');
    expect(wrapper.find('p').first().render().text()).toEqual('Small Text1');
  });
});
