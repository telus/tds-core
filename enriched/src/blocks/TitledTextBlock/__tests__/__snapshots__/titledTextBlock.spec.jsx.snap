exports[`TextBlock Component matches snapshot 1`] = `
<TitledTextBlock
  content={
    Array [
      Object {
        "content": "Small Text1",
        "title": "Small Title 1",
      },
    ]
  }
  title="Title1"
  titleTagType="h1">
  <div
    className="titled-text-block">
    <div
      className="container container--limited-width">
      <div
        className="grid-row">
        <div
          className="small-12 medium-12 large-12">
          <h1>
            Title1
          </h1>
          <div
            className="titled-text-component">
            <h4>
              Small Title 1
            </h4>
            <p>
              Small Text1
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</TitledTextBlock>
`;
